File: articles/water-energy-material-conditions-of-life/r/osmotic_pressure.R
# Osmotic pressure workflow in R.

solute_path <- file.path("data", "solute_conditions.csv")

if (!file.exists(solute_path)) {
  solute_path <- file.path("..", "data", "solute_conditions.csv")
}

solute <- read.csv(solute_path)

R_gas <- 0.082057

solute$osmotic_pressure_atm <-
  solute$van_t_hoff_factor *
  solute$concentration_mol_L *
  R_gas *
  solute$temperature_K

solute$relative_water_stress <-
  solute$osmotic_pressure_atm / max(solute$osmotic_pressure_atm)

print(round(solute, 5))

File: articles/water-energy-material-conditions-of-life/python/osmotic_pressure.py
"""
Osmotic pressure workflow.

Run:
    python python/osmotic_pressure.py
"""

from __future__ import annotations

from pathlib import Path

import pandas as pd

from water_energy_core import osmotic_pressure_atm


ARTICLE_DIR = Path(__file__).resolve().parents[1]
SOLUTE_PATH = ARTICLE_DIR / "data" / "solute_conditions.csv"


def main() -> None:
    """Calculate osmotic pressure across solute conditions."""

    df = pd.read_csv(SOLUTE_PATH)

    df["osmotic_pressure_atm"] = [
        osmotic_pressure_atm(i, c, t)
        for i, c, t in zip(
            df["van_t_hoff_factor"],
            df["concentration_mol_L"],
            df["temperature_K"],
        )
    ]

    df["relative_water_stress"] = df["osmotic_pressure_atm"] / df["osmotic_pressure_atm"].max()

    print(df.round(5).to_string(index=False))


if __name__ == "__main__":
    main()
